x/gov/keeper/proposal.go

```diff
@@ -110,26 +111,110 @@ func (keeper Keeper) GetProposal(ctx sdk.Context, proposalID uint64) (v1.Proposa
 		panic(err)
 	}
 
+	keeper.PopulateProposalContents(ctx, &proposal)
+
 	return proposal, true
 }
 
-// SetProposal sets a proposal to store.
+// GetProposalWithoutContents gets a proposal from store by ProposalID without messages and
+// metadata. Useful to reduce gas usage.
+// Panics if can't unmarshal the proposal.
+func (keeper Keeper) GetProposalWithoutContents(ctx sdk.Context, proposalID uint64) (v1.Proposal, bool) {
+	store := ctx.KVStore(keeper.storeKey)
+
+	bz := store.Get(types.ProposalKey(proposalID))
+	if bz == nil {
+		return v1.Proposal{}, false
+	}
+
+	var proposal v1.Proposal
```

Comment:
proposal still has the static data that we are trying to avoid, am I missing something?

Reply:
I guess what is the bare minimum needed for a vote? do we need the entire struct of proposal other than these two items?

Reply:
Are you asking if we could move more data into the other store? I think we could move these too:

```
  google.protobuf.Timestamp submit_time               = 5 [(gogoproto.stdtime) = true];
  google.protobuf.Timestamp deposit_end_time          = 6 [(gogoproto.stdtime) = true];
  google.protobuf.Timestamp         voting_start_time = 8 [(gogoproto.stdtime) = true];
  google.protobuf.Timestamp         voting_end_time   = 9 [(gogoproto.stdtime) = true];
```

Or what did you mean?

Reply:
> I guess what is the bare minimum needed for a vote?

MsgVote only needs to know the if proposal is in voting period or not, see [here](https://github.com/cosmos/cosmos-sdk/blob/main/x/gov/keeper/vote.go#L18)

Reply:
I would create a new type with the bare minimum needed for voting and use that. Current proposal struct stays the same

Reply:
Yeah the more I think about it, the more I think @tac0turtle brings up a good point. Honestly, this probably warrants a very short ADR. Describe the motivation, the new type(s) and the state changes. Can be super short. Just so we can agree on a direction and not waste time on a PR that isn't agreed upon to not waste @facundomedica's time.

Reply:
@alexanderbez imo the new struct only needs to contain a bool (IsInVotingPeriod?), hence my proposal to not even introduce a new struct https://github.com/cosmos/cosmos-sdk/pull/13576#discussion_r1000725690

I don't think this needs an adr though. Even the cancel proposal PR seems to me like a bigger change.

Reply:
That SGTM. I was just making a suggestion. If it's in good shape, let's proceed 👍